### PR TITLE
loan request UI fix

### PR DESCRIPTION
**Pull request by skyrag**
put the ui of the loan request and the item view creat and edit up to figma because all those screen use the same screen so everything is change at the same time

### CODE REVIEW

**Review by KassFlute — CHANGES_REQUESTED** [review]
Nice PR, I found some issues that should be addressed before merging to have nice loan requests

### UI notes:
- Not having the borrow item button on our own items is great but I think it looks strange to just hide the button. (see below)
Capture d’écran 2024-05-18 à 17 43 04

Maybe we could just make the button unclickable and tint it gray or something like this.

- I also feel that the font is really small on the view item screen (see code notes). 

### Code notes:
- Many text styles/colors/fonts are set but not by using the text style in the theme. These should be used. 
- Same for the colors, colors from the theme should be used. Now in darktheme a big part of the loan request design is unreadable/unusable. 
- The Loan screens should be tested in dark mode and on the small phone for the padding.
- Good code coverage

File: app/src/main/java/com/android/partagix/ui/screens/BorrowScreen.kt [hunk 1]
```diff
@@ -152,7 +159,7 @@ fun BorrowScreen(
                     label = { Text("Description") },
                     modifier = modifier.fillMaxWidth().testTag("description"),
                     minLines = 5,
-                    readOnly = false)
+                    readOnly = true)
```
[the remark this answers is not in the document]
Comment (skyrag) [reply app/src/main/java/com/android/partagix/ui/screens/BorrowScreen.kt:162]: i understood it as the description of the item and i don't understand why the description of the item could be change by the person borrowing it

[the remark this answers is not in the document]
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/BorrowScreen.kt:165]: I thought it was a description of the loan.

**Review by KassFlute — CHANGES_REQUESTED** [review]
- The grayed out borrow button on our own items is great
- The paddings are good on different phone sizes

But the colors are still wrong. You used theme's colors and that's great but you need to test with the dark mode enabled. See below the loan screen in dark mode that is strange.
Capture d’écran 2024-05-19 à 10 32 21

**Review by KassFlute — APPROVED** [review]
There is still some colors that are wrong. For example you use "onPrimary" for buttons that are on the background. "onPrimary" should be used for elements that are on the color "primary".
But it's ok we can make future other tasks to change the colors.

File: app/src/main/java/com/android/partagix/ui/screens/BorrowScreen.kt [hunk 2]
```diff
@@ -270,6 +273,12 @@ fun BorrowScreen(
 
               Button(
                   modifier = modifier.fillMaxWidth().testTag("saveButton").padding(10.dp),
+                  colors =
+                      ButtonColors(
+                          containerColor = MaterialTheme.colorScheme.onSecondaryContainer,
+                          contentColor = MaterialTheme.colorScheme.background,
```
[the remark this answers is not in the document]
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/BorrowScreen.kt]: still wrong but more acceptable as it's readable in both modes

**Pull request merged**